### PR TITLE
feat(pod): allow adding an existing `Container` object

### LINKED ISSUE #1508: Deployment doesn't accept Container objects
Hi
I'm following this [doc](https://pypi.org/project/cdk8s-plus/) and trying to use `Container.mount()`

The problem is that the `Deployment.containters` expects `ContainerProps`, but `ContainerProps` doesn't have `mount` method.

The only way to use the mount method is by getting the container object from `Deployment.containers` after the deployment is created.

So the given example doesn't work
```python
container = kplus.Container(
    image="node:14.4.0-alpine3.12",
    command=["node", "index.js", f"{port}"],
    port=port,
)

container.mount(app_path, app_volume)
deployment = kplus.Deployment(chart, "Deployment",
    replicas=3,
    containers=[container]
)

``` 
Results with 
```
TypeError: type of argument containers must be one of (Sequence[Union[cdk8s_plus_25.ContainerProps, Dict[str, Any]]], NoneType); got list instead
```

The way to do this is
```python
container_p = kplus.ContainerProps(
    image="node:14.4.0-alpine3.12",
    command=["node", "index.js", f"{port}"],
    port=port,
)

deployment = kplus.Deployment(chart, "Deployment",
    replicas=3,
    containers=[container_p]
)

container = deployment.containers[0]
container.mount(app_path, app_volume)
```

This would be easier if Deployment would accept Container, or if Container had a `.props` property to convert it

### PR DESCRIPTION
**Pull request by yfried**
Fixes: #1508

### CODE REVIEW

**Review by iliapolo — CHANGES_REQUESTED** [review]
@yfried Unfortunately we can't do this because union types don't translate well to other languages. In Java for example this would translate to accepting an `Object`, which is not a good experience. What we usually do is just add another method with a different name that accepts the different argument.

**Comment by yfried** [comment]
> @yfried Unfortunately we can't do this because union types don't translate well to other languages. In Java for example this would translate to accepting an `Object`, which is not a good experience. What we usually do is just add another method with a different name that accepts the different argument.

So instead you meant to have `attachConainer(cont: Container)` and `addContainer(cont: ContainerProps)`?
It's easy to do technically but I'm concerned that these 2 methods will look the same to new users (like me), and would raise a whole new bunch of questions/frustrations

**Comment by iliapolo** [comment]
@yfried I agree adding another method might be confusing, but we can clarify the confusion with documentation. 

Ideally we would have the `addContainer` method accept a `Container`, and then have another method called `createContainer` that would accept `ContainerProps` and create the `Container`. But unfortunately we cannot break the API.

So given our limitations, I think adding the `attachXXX` methods will be an OK solution.

**Review by iliapolo — CHANGES_REQUESTED** [review]
@yfried Lets also add `attachInitContainer`, and we need some unit tests added here: https://github.com/cdk8s-team/cdk8s-plus/blob/k8s-25/main/test/pod.test.ts#L138

**Review by iliapolo — APPROVED** [review]

**Comment by iliapolo** [comment]
@yfried looks good. Just need the DCO: https://github.com/cdk8s-team/cdk8s-plus/blob/k8s-25/main/CONTRIBUTING.md#developer-certificate-of-origin-dco

File: src/pod.ts [hunk 1]
```diff
@@ -107,10 +107,14 @@ export abstract class AbstractPod extends base.Resource implements IPodSelector,
 
   public addContainer(cont: container.ContainerProps): container.Container {
     const impl = new container.Container(cont);
-    this._containers.push(impl);
+    this.attachContainer(impl)
```
Comment (iliapolo) [inline src/pod.ts]:
```suggestion
    this.attachContainer(impl);
```

**Comment by iliapolo** [comment]
@yfried The build is failing because it detected a diff between your code and the one our linters expect. Can you run `yarn build` locally to auto-fix it?

**Pull request merged**